### PR TITLE
feat(GRO-1113): User sees app path for `Follow artists I’m interested in`

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [GRO-1113]

### Description

- adds follow artists screen
- dynamic to easily support the galleries follows afterwards
- added some dumb tests (will add more on the galleries pr)
- collector can search screen
- title goes away when collector starts typing
- button disabled when no followed artists

> Missing the header and progress bar right now, will be added after it is finished in another ticket. Also todo: add animation when reanimated2 is merged to hide header dynamically

cc @artsy/grow-devs 

#### demo vid

https://user-images.githubusercontent.com/21178754/181746069-3bd6e482-f261-4759-974e-72d288471034.mp4

### QA Test Case(s)

| Test Case | Feature | Environment | Acceptance Criteria | Setup Instructions/Link |
| --------- | :-----: | ----------: | ------------------: | ----------------------: |
|           |         |             |                     |                         |

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [x] I hid my changes behind a [feature flag].

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- add follow artists screen new onboarding - gkartalis

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[GRO-1113]: https://artsyproduct.atlassian.net/browse/GRO-1113?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/shared/hooks/useDebouncedValue.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { debounce } from "lodash"
+import { useCallback, useEffect, useState } from "react"
+
+interface UseDebounce {
+  delay?: number
+  callback: (...args: any[]) => void
+}
+
+export const useDebounce = ({ callback, delay = 200 }: UseDebounce) => {
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (olerichter00) [inline src/shared/hooks/useDebouncedValue.ts:10]: This should work without disabling the rule.

```suggestion
```

File: src/shared/hooks/useDebouncedValue.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  return useCallback(debounce(callback, delay), [callback, delay])
+}
+
+interface UseDebouncedValue<T> {
+  value: T
+  delay?: number
+}
+
+export const useDebouncedValue = <T>({ value, delay = 200 }: UseDebouncedValue<T>) => {
```
Comment (olerichter00) [inline src/shared/hooks/useDebouncedValue.ts:19]: Nice!

File: src/app/Scenes/Onboarding/OnboardingPersonalization2/ArtistListItem.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    </Flex>
+  )
+}
+
+const FollowArtistMutation = graphql`
+  mutation ArtistListItemNewFollowArtistMutation($input: FollowArtistInput!) {
+    followArtist(input: $input) {
+      artist {
+        id
+        is_followed: isFollowed
```
Comment (olerichter00) [inline src/app/Scenes/Onboarding/OnboardingPersonalization2/ArtistListItem.tsx:56]: Is this needed or can we just change it to

```suggestion
        isFollowed
```

?

**Review by olerichter00 — APPROVED** [review]
Looks great!

File: src/shared/hooks/useDebouncedValue.ts (hunk 1 above)
Comment (gkartalis) [reply src/shared/hooks/useDebouncedValue.ts:10]: indeed, copied it over from force :D

File: src/app/Components/ArtistListItem.tsx [hunk 4]
```diff
@@ -228,7 +228,7 @@ export const ArtistListItemPlaceholder = () => (
       <PlaceholderText height={13} width={100 + Math.random() * 100} />
     </Flex>
     <Flex height={45} position="absolute" right={0} justifyContent="center">
-      <PlaceholderBox height={25} width={70} borderRadius={2} />
+      <PlaceholderBox height={25} width={90} borderRadius={50} />
```
Comment (gkartalis) [inline src/app/Components/ArtistListItem.tsx:231]: this represents the follow button, adjusted to match the button style

File: src/app/Scenes/Onboarding/OnboardingPersonalization2/OnboardingOrderedSet.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+const OnboardingOrderedSet: React.FC<OnboardingOrderedSetProps> = ({ id }) => {
+  const { orderedSets } = useLazyLoadQuery<OnboardingOrderedSetQuery>(
+    OnboardingOrderedSetScreenQuery,
+    {
+      key: id,
+    }
+  )
+  const space = useSpace()
```
Comment (damassi) [inline src/app/Scenes/Onboarding/OnboardingPersonalization2/OnboardingOrderedSet.tsx:23]: haha wow, this makes a lot of sense. We've been using `themeGet` from a lib which is pretty meh

**Review by damassi — APPROVED** [review]
Looks just about perfect

**Pull request merged**